nomos-services/blend/src/lib.rs

```diff
@@ -117,89 +126,306 @@ where
 
-        let membership_adapter = <MembershipAdapter<EdgeService> as membership::Adapter>::new(
+        let _membership_stream = <MembershipAdapter<EdgeService> as membership::Adapter>::new(
             overwatch_handle
                 .relay::<MembershipService<EdgeService>>()
                 .await?,
             settings.crypto.signing_private_key.public_key(),
-        );
-        let mut _session_stream = SessionEventStream::new(
-            membership_adapter.subscribe().await?,
+        )
+        .subscribe()
+        .await?;
+        // TODO: Use membership_stream once the membership/SDP services are ready to provide the real membership: https://github.com/logos-co/nomos/issues/1532
+
+        let mut membership = settings.membership();
+        let mut session_stream = SessionEventStream::new(
+            Box::pin(constant_membership_stream(
+                membership.clone(),
+                settings.time.session_duration(),
+            )),
             settings.time.session_transition_period(),
         );
-        // TODO: Use session_stream: https://github.com/logos-co/nomos/issues/1532
-
-        // TODO: Add logic to start/stop the core or edge service based on the new
-        // membership info.
-
-        let minimal_network_size = settings.minimal_network_size;
-        let membership_size = settings.membership.len();
-
-        if membership_size >= minimal_network_size.get() as usize {
-            wait_until_services_are_ready!(
-                &overwatch_handle,
-                Some(Duration::from_secs(60)),
-                CoreService,
-                MembershipService<EdgeService>
-            )
-            .await?;
-            let core_relay =
-                overwatch_handle
-                .relay::<CoreService>()
-                .await?;
-            status_updater.notify_ready();
-            info!(
-                target: LOG_TARGET,
-                "Service '{}' is ready.",
-                <RuntimeServiceId as AsServiceId<Self>>::SERVICE_ID
-            );
-            while let Some(message) = inbound_relay.next().await {
-                debug!(target: LOG_TARGET, "Relaying a message to core service");
-                if let Err((e, _)) = core_relay.send(message).await {
-                    error!(target: LOG_TARGET, "Failed to relay message to core service: {e:?}");
-                }
-            }
-        } else {
-            wait_until_services_are_ready!(
-                &overwatch_handle,
-                Some(Duration::from_secs(60)),
-                NetworkService<NetworkBackendOfService<CoreService, RuntimeServiceId>, _>,
-                MembershipService<EdgeService>
-            )
-            .await?;
-            let core_network_relay =
-                overwatch_handle
+
+        // Relays to the core and edge services.
+        let core_relay = overwatch_handle.relay::<CoreService>().await?;
```

Comment:
Why are we creating relays with core and edge services? These should exist only if the network is large enough, else it should not be possible to interact with them and the compiler should prevent that, which for me means they should be optional and instantiated if a new session is large enough, or deconstructed if the new session is too small.